cdap-kubernetes/src/main/java/co/cask/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -21,44 +21,59 @@
 import co.cask.cdap.master.spi.environment.MasterEnvironmentContext;
 import org.apache.twill.discovery.DiscoveryService;
 import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.io.BufferedReader;
 import java.io.File;
 import java.io.IOException;
-import java.io.Reader;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.util.HashMap;
 import java.util.Map;
-import java.util.Properties;
 import java.util.function.Supplier;
-import java.util.stream.Collectors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * Implementation of {@link MasterEnvironment} to provide the environment for running in Kubernetes.
  */
 public class KubeMasterEnvironment implements MasterEnvironment {
 
+  private static final Logger LOG = LoggerFactory.getLogger(KubeMasterEnvironment.class);
+
   private static final String NAMESPACE_KEY = "master.environment.k8s.namespace";
   private static final String POD_LABELS_PATH = "master.environment.k8s.pod.labels.path";
 
   private static final String DEFAULT_NAMESPACE = "default";
   private static final String DEFAULT_POD_LABELS_PATH = "/etc/podinfo/pod.labels.properties";
 
+  private static final Pattern LABEL_PATTERN = Pattern.compile("(cdap\\..+?)=\"(.*)\"");
```

Comment:
wouldn't this always be `cdap.service=`?

Reply:
no. a pod can have multiple labels.